### PR TITLE
Add `.cumulative` to `cumsum` & `cumprod` docstrings

### PR DESCRIPTION
**Pull request by max-sixty**
As discussed in a couple of issues, we should be directing folks towards the `.cumulative`. (the only missing piece is `skip_na`...

I thought this  was a reasonable way to have the generation script work for these; ofc open to feedback.

I also added the namedarray file to the instructions for generating

### CODE REVIEW

**Comment by dcherian** [comment]
> Note that the methods on the ``cumulative`` method are more performant  and better supported

I'm surprised they're more performant than using numpy's cumprod, cumsum directly. Is this because `cumulative` redirects to numbagg?

**Comment by max-sixty** [comment]
Yes! :)

I don't have benchmarks at https://github.com/numbagg/numbagg, since `cumulative().sum()` just calls `move_sum` with the appropriate window, and numpy doesn't have a comparable function.

But to confirm — 10x faster one over 10 columns, 2x faster over 1 column:

```python
[nav] In [36]: A = np.random.rand(60000, 10)

[ins] In [40]: %timeit np.cumsum(A, axis=0)
2.44 ms ± 82.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

[ins] In [47]: %timeit numbagg.move_sum(A, window=60000, min_count=0, axis=0)
371 µs ± 16.5 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

```python
A = np.random.rand(60000, 1)

[ins] In [49]: %timeit numbagg.move_sum(A, window=60000, min_count=0, axis=0)
106 µs ± 1.62 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

[nav] In [50]: %timeit np.cumsum(A, axis=0)
211 µs ± 5.34 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

**Comment by dcherian** [comment]
we should clarify that then!

for the equivalent non-numbagg benchmark you could use xarray with `use_numbagg=False`?

**Comment by max-sixty** [comment]
> we should clarify that then!

In the `.cumulative` docstring?

File: xarray/core/_aggregations.py [hunk 1]
```diff
@@ -1069,13 +1069,18 @@ def cumsum(
...
         DataArray.cumsum
+        Dataset.cumulative
         :ref:`agg`
             User guide on reduction or aggregation operations.
 
         Notes
         -----
         Non-numeric variables will be removed prior to reducing.
 
+        Note that the methods on the ``cumulative`` method are more performant
```
Comment (dcherian) [inline xarray/core/_aggregations.py:1080]:
```suggestion
        Note that the methods on the ``cumulative`` method are more performant with numbagg installed
```

File: xarray/util/generate_aggregations.py [hunk 2]
```diff
@@ -223,6 +223,9 @@ def {method}(
 See the `flox documentation <https://flox.readthedocs.io>`_ for more."""
 _FLOX_GROUPBY_NOTES = _FLOX_NOTES_TEMPLATE.format(kind="groupby")
 _FLOX_RESAMPLE_NOTES = _FLOX_NOTES_TEMPLATE.format(kind="resampling")
+_CUM_NOTES = """Note that the methods on the ``cumulative`` method are more performant
```
Comment (dcherian) [inline xarray/util/generate_aggregations.py:226]:
```suggestion
_CUM_NOTES = """Note that the methods on the ``cumulative`` method are more performant (with numbagg installed)
```

**Comment by max-sixty** [comment]
Merged, also improved the docs & suggested commands in `generate_aggregations.py`

**Pull request merged**